packages/valory/skills/registration_abci/behaviours.py

```diff
@@ -361,43 +361,44 @@ def restart_tendermint(self) -> Generator[None, None, bool]:
     def async_act(self) -> Generator:
         """Act asynchronously"""
 
-        self.context.logger.info(f"My address: {self.context.agent_address}")
-        yield from self.sleep(self.params.sleep_time)
-
-        # collect personal Tendermint configuration
-        if not self.local_tendermint_params:
-            successful = yield from self.get_tendermint_configuration()
-            if not successful:
-                yield from self.sleep(self.params.sleep_time)
-                return
-
-        # make service registry contract call
-        if not self.registered_addresses:
-            successful = yield from self.get_addresses()
-            if not successful:
-                yield from self.sleep(self.params.sleep_time)
-                return
-
-        # collect Tendermint config information from other agents
-        if not self.collection_complete:
-            successful = yield from self.request_tendermint_info()
-            if not successful:
-                yield from self.sleep(self.params.sleep_time)
-                return
+        if self.params.share_tm_config_on_startup:
+            self.context.logger.info(f"My address: {self.context.agent_address}")
+            yield from self.sleep(self.params.sleep_time)
```

Comment:
yes that is why. If no, then you don't need to scroll further even you see immediately where you go next, otherwise you need to scroll over the indentation block to see what's next

Reply:
Yes I agree with @Karrenbelt , always try to use guard statements that return as early as possible. It makes code more easy to read and avoid nesting.

Reply:
Done in 3b696c961d43bd8aedbe7d9dea7f118a175580ea, though I would prefer to create a `share_tm_config` method for the nested part.

Reply:
great idea. Do it.